EventFilter/SiStripRawToDigi/interface/SiStripFEDBuffer.h

```diff
@@ -130,6 +134,35 @@ namespace sistrip {
       uint16_t valuesLeft_;
     };
 
+  //class for unpacking data from any FED channels with a non-integer words bits stripping mode
+  class FEDBSChannelUnpacker
+    {
+    public:
+      static FEDBSChannelUnpacker virginRawModeUnpacker(const FEDChannel& channel, size_t num_bits);
+      static FEDBSChannelUnpacker zeroSuppressedModeUnpacker(const FEDChannel& channel, size_t num_bits);
+      static FEDBSChannelUnpacker zeroSuppressedLiteModeUnpacker(const FEDChannel& channel, size_t num_bits);
+      FEDBSChannelUnpacker();
+      uint8_t sampleNumber() const;
+      uint16_t adc() const;
+      bool hasData() const;
+      FEDBSChannelUnpacker& operator ++ ();
+      FEDBSChannelUnpacker& operator ++ (int);
+    private:
+      //pointer to beginning of FED or FE data, offset of start of channel payload in data and length of channel payload
+      FEDBSChannelUnpacker(const uint8_t* payload, const size_t channelPayloadOffset, const int16_t channelPayloadLength, const size_t offsetIncrement=10);
+      static void throwBadChannelLength(const uint16_t length);
+      static void throwBadWordLength(const size_t word_length);
+      const uint8_t* data_;
+      size_t oldWordOffset_;
```

Comment:
size_t changes between architectures.
It's 64 now in most of what we run.
Does it matter in the algorithm logic?